Mathlib/MeasureTheory/Measure/Lebesgue/EqHaar.lean

```diff
@@ -79,6 +83,42 @@ theorem Basis.parallelepiped_basisFun (ι : Type*) [Fintype ι] :
     · exact zero_le_one
 #align basis.parallelepiped_basis_fun Basis.parallelepiped_basisFun
 
+ /-- A parallelepiped can be expressed on the standard basis. -/
+ theorem Basis.parallelepiped_eq_map  {ι E : Type*} [Fintype ι] [NormedAddCommGroup E]
+    [NormedSpace ℝ E] (b : Basis ι ℝ E) :
+    b.parallelepiped = (TopologicalSpace.PositiveCompacts.piIcc01 ι).map b.equivFun.symm
+    b.equivFunL.symm.continuous
+      (by have := FiniteDimensional.of_fintype_basis b
+          exact LinearMap.isOpenMap_of_finiteDimensional _ b.equivFun.symm.surjective) := by
+  rw [← Basis.parallelepiped_basisFun]
+  refine (congrArg _ ?_).trans (Basis.parallelepiped_map _ _)
+  classical
+  ext
+  simp only [map_apply, Pi.basisFun_apply, equivFun_symm_apply, LinearMap.stdBasis_apply',
+    Finset.sum_univ_ite]
+
+theorem Basis.parallelepiped_measurable {ι E : Type*} [Fintype ι] [NormedAddCommGroup E]
+    [NormedSpace ℝ E] [MeasurableSpace E] [OpensMeasurableSpace E] (b : Basis ι ℝ E) :
+    MeasurableSet (b.parallelepiped : Set E) := by
+  rw [Basis.parallelepiped_eq_map, PositiveCompacts.coe_map, LinearEquiv.image_symm_eq_preimage]
+  exact (continuous_equivFun_basis _).measurable (PositiveCompacts.piIcc01_measurable ι)
+
+open MeasureTheory MeasureTheory.Measure
+
+theorem Basis.addHaar_map {ι E F : Type*} [Fintype ι] [NormedAddCommGroup E]
```

Comment:
I think `map_addHaar` might be a better name here, since the statement is `map _ (addHaar b) = _`. I'd also be fine with switching the direction of the statement instead.

Of course, maybe your name is consistent with something earlier in the file; if so, you're off the hook and this is fine!